cmd/api/src/api/v2/saved_queries.go

```diff
@@ -89,9 +89,47 @@ func (s Resources) ListSavedQueries(response http.ResponseWriter, request *http.
 			api.WriteErrorResponse(request.Context(), ErrBadQueryParameter(request, model.PaginationQueryParameterSkip, err), response)
 		} else if limit, err := ParseLimitQueryParameter(queryParams, 10000); err != nil {
 			api.WriteErrorResponse(request.Context(), ErrBadQueryParameter(request, model.PaginationQueryParameterLimit, err), response)
-		} else if queries, count, err := s.DB.ListSavedQueries(request.Context(), user.ID, strings.Join(order, ", "), sqlFilter, skip, limit); err != nil {
-			api.HandleDatabaseError(request, response, err)
+		} else if len(scopes) == 0 {
+			if queries, count, err := s.DB.ListSavedQueries(request.Context(), user.ID, strings.Join(order, ", "), sqlFilter, skip, limit); err != nil {
+				api.HandleDatabaseError(request, response, err)
+			} else {
+				api.WriteResponseWrapperWithPagination(request.Context(), queries, limit, skip, count, http.StatusOK, response)
+			}
 		} else {
+			var queries []model.SavedQueryResponse
+			var count int
+			for _, scope := range strings.Split(scopes[0], ",") {
+				var scopedQueries model.SavedQueries
+				var scopedCount int
+
+				switch scope {
+				case "public":
+					scopedQueries, err = s.DB.GetPublicSavedQueries(request.Context())
+					scopedCount = len(scopedQueries)
+				case "shared":
+					scopedQueries, err = s.DB.GetSharedSavedQueries(request.Context(), user.ID)
+					scopedCount = len(scopedQueries)
+				case "owned":
+					scopedQueries, scopedCount, err = s.DB.ListSavedQueries(request.Context(), user.ID, strings.Join(order, ", "), sqlFilter, skip, limit)
```

Comment:
If these are the same scopes as https://github.com/SpecterOps/BloodHound/blob/3a68a2240e716d4fd9a89fad77c014d7d03cee78/cmd/api/src/database/saved_queries_permissions.go

We should use the constants as those are Title case and these are lowercase

Reply:
They are the same! I didn't know those had been added in Ari's PR.  Since her PR isn't merged yet, would you like me to go ahead and add them in mine? Or wait until her PR has been merged?

Reply:
You may want to create separate constants in this file, as they are the same, but different (title vs lowercase, and different layers). I guess we could use the same constants if we were to `toLower` the constant before checking it against scope (may want to `toLower` the scope var in the switch too)

Reply:
Actually, reviewing those constants, I'm not sure they should be title case. We could probably get those changed to lowercase and reuse them. It wouldn't hurt to make sure we're setting the incoming `scope` parameter to lowercase before checking against our strings/constants either way though

Reply:
Updated this.  Let me know what you both think.  I did just add the consts that are in Ari's PR.

Reply:
This is great. Ship it.